openmdao/utils/options_dictionary.py

```diff
@@ -148,6 +148,19 @@ def declare(self, name, default=_undefined, values=None, types=None, type_=None,
         if default_provided:
             self._assert_valid(name, default)
 
+    def undeclare(self, name):
+        """
+        Remove entry from the OptionsDictionary, for classes that don't use that option.
+
```

Comment:
You should add a unit test to `test_options_dictionary` that verifies that removing a key removes it (i.e., it should raise a keyerror if you try to interact with the removed option.)

Reply:
Done, @Kenneth-T-Moore ... see what you think.